impl/src/checks.rs

```diff
@@ -51,23 +88,54 @@ pub fn check_all_vendored(
   }
 
   // Oops, missing some crates. Yield a nice message
-  let expected_full_path = env::current_dir().unwrap().join(VENDOR_DIR);
+  let expected_full_path = vendor_path(
+    bazel_workspace_root,
+    &settings.workspace_path,
+    &settings.vendor_dir,
+  );
 
   Err(
     RazeError::Planning {
       dependency_name_opt: None,
       message: format!(
         "Failed to find expected vendored crates in {:?}: {:?}. Did you forget to run cargo \
          vendor?",
-        expected_full_path.to_str(),
+        expected_full_path.display(),
         limited_missing_crates
       ),
     }
     .into(),
   )
 }
 
-pub fn check_resolve_matches_packages(metadata: &Metadata) -> Result<()> {
+fn vendor_path(bazel_workspace_root: &Path, workspace_path: &str, vendor_dir: &str) -> PathBuf {
+  bazel_workspace_root
+    // Trim the absolute label identifier from the start of the workspace path
+    .join(workspace_path.trim_start_matches('/'))
+    .join(vendor_dir)
+}
+
+/**
+ * Returns the packages expected path during current execution.
+ *
+ * Not for use except during planning as path is local to run location.
```

Comment:
I can't tell if this is a new comment or not, but you're literally using this function outside of planning because this is called by `check_all_vendored`, which is called by `check_metadata`, which is called in cargo-raze.rs. Delete this comment?